### ISSUE TITLE
node: Using native modules requires running with dev image

### ISSUE THREAD

**Issue by ligustah** [iron-io/dockers#36]
Hi,

I've been trying to use the node image for a project of mine and set up a workflow where `iron/node:5-dev` is used to run `npm install` before building the images based on `iron/node:5`. When doing this with a project that depends on native node modules I get this error when trying to use these images:
```
Error: Dynamic loading not supported
    at Error (native)
    at Object.Module._extensions..node (module.js:440:18)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at bindings (/app/node_modules/bindings/bindings.js:76:44)
    at Object.<anonymous> (/app/node_modules/contextify/lib/contextify.js:1:96)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
```
They do work perfectly fine when running using a dev image. Apparently this is because the non-dev images are build with `--fully-static`. I'm not sure if you can get small images without that option, but you might want to add a note to your documentation stating that this problem exists.

**Comment by treeder** [comment]
Hmm, maybe due to using someone else's images for 5 temporarily. 

Could you try the iron/node:4 images?

**Comment by ligustah** [comment]
Works with iron/node:4

**Comment by andrest** [comment]
I'm facing a similar error. Getting `Dynamic loading not supported` with iron/node:5. iron/node:4 seems to work, but still getting an application level error that I can't replicate locally though. The latter might be unrelated to the question in hand, but just so it's not, I'm getting a persistent "getaddrinfo EAI_AGAIN" error when trying to connect to a remote database.

@treeder what is the difference between iron/node image versions?

**Comment by treeder** [comment]
@andrest are you on windows?  http://stackoverflow.com/questions/30702702/eai-again-when-installing-express-from-npm-in-windows-7

And regarding the difference, iron/node:5 is based off this: https://hub.docker.com/r/mhart/alpine-node/ while iron/node:4 is a normal install. I'd recommend using iron/node:4 for now though as it seems to have less issues.

**Comment by ligustah** [comment]
@andrest As I mentioned in the initial description, the (apparently) important difference is the [fully static option](https://github.com/mhart/alpine-node/blob/df28b18177d42ae0d1c7678672760b30bf82d139/Dockerfile#L10) used by the base image for iron/node:5.

**Comment by treeder** [comment]
Suppose we could just build this on our own, non-static....

**Comment by treeder** [comment]
Could you guys try `iron/node:5-dev-test` and `iron/node:5-test`. If they work, I'll make those the images we use going forward.

**Comment by ligustah** [comment]
I cannot currently test this against the original project I was working on, but I created a simple test case that shows that dynamic loading is working as expected with the new image.

See https://gist.github.com/ligustah/2acf759e2dc3ccac8027

I called it like this: `./test-iron-node.sh` to verify the failure with the old image. Then `test-iron-node.sh -test` to check the new images.

Thanks for the fix!
